### PR TITLE
add 6.6 release notes

### PR DESCRIPTION
**Pull request by pcsanwald**
Add release notes for 6.6. Issuing a PR so I can get a review before merging since this is my first time generating release notes for ES.

### CODE REVIEW

File: docs/reference/release-notes/6.6.asciidoc [hunk 1]
```diff
@@ -1,52 +1,646 @@
...
-// === Breaking Changes
+[[breaking-6.6.0]]
+[float]
+=== Breaking changes
 
-// [float]
-// [[breaking-java-n.n.n]]
-// === Breaking Java Changes
+Authentication::
+* Enhance Invalidate Token API {pull}35388[#35388] (issues: {issue}34556[#34556], {issue}35115[#35115])
```
Comment (lcawl) [inline docs/reference/release-notes/6.6.asciidoc]: This item should be removed per discussion with @jaymode (the "breaking" label was removed just a few moments ago, likely after you ran this script)

File: docs/reference/release-notes/6.6.asciidoc [hunk 2]
```diff
@@ -1,91 +1,635 @@
...
-// To add a release, copy and paste the following text,  uncomment the relevant
-// sections, and add a link to the new section in the list of releases in
-// ../release-notes.asciidoc. Note that release subheads must be floated and
-// sections cannot be empty.
-// TEMPLATE
 
-// [[release-notes-n.n.n]]
-// == {es} version n.n.n
+[[release-notes-6.6.0]]
+== 6.6.0 Release Notes
```
Comment (lcawl) [inline docs/reference/release-notes/6.6.asciidoc]: To match the other existing pages:

```suggestion
== {es} version 6.6.0
```

File: docs/reference/release-notes/6.6.asciidoc [hunk 3]
```diff
@@ -1,91 +1,635 @@
...
 
-// coming[n.n.n]
+coming[6.6.0]
```
Comment (lcawl) [inline docs/reference/release-notes/6.6.asciidoc]: The docs are frozen, so this is not required:

```suggestion

```

File: docs/reference/release-notes/6.6.asciidoc [hunk 4]
```diff
@@ -1,91 +1,621 @@
...
+Distributed::
+* Allow asynchronous block operations to be delayed in IndexShardOperationPermits {pull}35999[#35999] (issues: {issue}35540[#35540], {issue}35850[#35850])
+* TransportResyncReplicationAction should not honour blocks {pull}35795[#35795] (issues: {issue}35332[#35332], {issue}35597[#35597])
+* Add global and index level blocks to IndexSettings {pull}35695[#35695] (issues: {issue}35332[#35332], {issue}35597[#35597])
+* Expose all permits acquisition in IndexShard and TransportReplicationAction {pull}35540[#35540] (issue: {issue}33888[#33888])
+* Add a java level freeze/unfreeze API {pull}35353[#35353] (issues: {issue}34352[#34352], {issue}34357[#34357])
+* Check blocks while having index shard permit in TransportReplicationAction {pull}35332[#35332] (issue: {issue}33888[#33888])
+* Apply masterNodeTimeout to MasterNodeRequest transmission {pull}35235[#35235]
+
+Docs Infrastructure::
```
Comment (lcawl) [inline docs/reference/release-notes/6.6.asciidoc]: I think this PR fits better in the HLRC category.

File: docs/reference/release-notes/6.6.asciidoc (hunk 1 above)
Comment (pcsanwald) [reply docs/reference/release-notes/6.6.asciidoc]: got it, I mis-interpreted @lcawl's comment. I've added it to the enhancements section in eea93cc77d4f01bfbd0131b9c3557925523bd025

**Review by lcawl — APPROVED** [review]
Other than potentially removing the "Build" sections, it LGTM

**Review by debadair — APPROVED** [review]

**Comment by pcsanwald** [comment]
@lcawl I'll follow up with a separate PR if we need to remove "build" section

**Pull request merged**